Improve error messages for slices

Here, the name of the element (Observation.value) is somewhere in the middle and repeated completely at the end.

```
[ERROR] Element is of type 'boolean', which is not one of the allowed choice types ('string','Quantity') (at [0].entry[2].resource[0].valueBoolean[0], element Bundle(http://example.org/StructureDefinition/DocumentBundle).entry.resource->Observation.value[x][@default])
```

Is this good enough?  Just make an exception for slices? Everywhere?

--- a/src/Firely.Fhir.Validation.Shared/Validator.cs
+++ b/src/Firely.Fhir.Validation.Shared/Validator.cs
@@ -99,7 +99,9 @@ namespace Firely.Fhir.Validation
 
             var validator = new SchemaReferenceValidator(profile);
             updateContext();
-            return validator.Validate(sn, _settings).ToOperationOutcome();
+            return validator.Validate(sn, _settings)
+                .CleanUp() // cleans up the error outcomes.
+                .ToOperationOutcome();
         }
     }
 }

